lib/CXGN/BrAPI/v2/ObservationVariables.pm

```diff
@@ -124,240 +145,261 @@ sub search {
 
     my $and_where_clause = join ' AND ', @and_wheres;
 
-    my @data;
+    $self->get_query($supported_crop, $and_where_clause, $join, 1);
+
+}
+
+sub detail {
+    my $self = shift;
+    my $trait_id = shift;
+    my $c = shift;
+    my $supported_crop = $c->config->{'supportedCrop'};
+
+    my $join = '';
+    my $and_where;
+
+    # only get cvterms for the cv for the configured trait ontology
+    $and_where = "cvterm.cv_id=".$self->trait_ontology_cv_id;
+
+    if ($trait_id){
+        $and_where = $and_where." AND cvterm.cvterm_id IN ($trait_id)";
+    }
+
+    $self->get_query($supported_crop, $and_where, $join, 0);
+
+}
+
+sub get_query {
+    my $self = shift;
+    my $supported_crop = shift;
+    my $and_where = shift;
+    my $join = shift;
+    my $array = shift;
+
+    #my $page_size = $self->page_size;
+    my $page_size = 1000; # ignore page size for now
```

Comment:
Is there a card to fix this when pagination is implemented in bi-api? Instead of hard coding this in breedbase, can we hard code pagination in bi-api on the request side? Might be easier to fix it there when we add pagination support.

Reply:
Added item about updating it in https://breedinginsight.atlassian.net/browse/BI-703?atlOrigin=eyJpIjoiNTYyMTVlODhjZWNkNGI2YjlmN2ViMGZiMjU3MDVhOTUiLCJwIjoiaiJ9. I just did it in breedbase because I was already working in there and was faster than going into bi-api. It may be worth doing that though for the rest of the integration work.